node/impl/full/eth.go

```diff
@@ -525,6 +530,20 @@ func (a *EthModule) EthGetStorageAt(ctx context.Context, ethAddr ethtypes.EthAdd
 		return nil, fmt.Errorf("failed to construct system sender address: %w", err)
 	}
 
+	actor, err := a.StateManager.LoadActor(ctx, to, ts)
+	if err != nil {
+		if xerrors.Is(err, types.ErrActorNotFound) {
+			return nil, nil
+		}
+		return nil, xerrors.Errorf("failed to lookup contract %s: %w", ethAddr, err)
+	}
+
+	// Not a contract. Technically, we should return an empty slot... but an error is more user
+	// friendly, IMO.
+	if !builtinactors.IsEvmActor(actor.Code) {
+		return nil, xerrors.Errorf("actor is not an ethereum contract")
+	}
```

Comment:
This will break libraries. Eth libraries expect a zero answer for anything that is not a smart contract OR for storage keys that don't exist.

Reply:
Got it.